MachineLearning_Engine/Compute/InstallToolkit.cs

```diff
@@ -0,0 +1,124 @@
+﻿/*
+ * This file is part of the Buildings and Habitats object Model (BHoM)
+ * Copyright (c) 2015 - 2020, the respective contributors. All rights reserved.
+ *
+ * Each contributor holds copyright over their respective contributions.
+ * The project versioning (Git) records all such contribution source information.
+ *                                           
+ *                                                                              
+ * The BHoM is free software: you can redistribute it and/or modify         
+ * it under the terms of the GNU Lesser General Public License as published by  
+ * the Free Software Foundation, either version 3.0 of the License, or          
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using BH.oM.Reflection;
+using System;
+using System.Collections.Generic;
+using System.IO;
+
+namespace BH.Engine.MachineLearning
+{
+    public static partial class Compute
+    {
+        /*************************************/
+        /**** Public Methods              ****/
+        /*************************************/
+
+        public static Output<bool, List<string>> InstallToolkit(bool force = false, bool run = false)
+        {
+            if (!run)
+                return null;
+
+            string module;
+            List<string> installed = new List<string>();
```

Comment:
On the Python Toolkit graphs branch @tg359 and I were working on we refactored this, to have the list of modules needing to be installed at the top, instantiated with the modules, then using a `foreach` loop to loop over each module and install it.
The joy of that is when we want to add another package we only need to update the initial list, without needing to copy/paste the install code for each new package.
What are your thoughts on that one @epignatelli ?

Reply:
Yeah, I had seen that, and I agree, the code is awful. 
The reason I didn't do it is that some packages have additional parameters in the `pip install`, such as version, or links.

I could group together the ones that are simple pip install and manually add the rest?

Reply:
Ah ok, I hadn't considered that. I am not against doing it this way, makes it very clear what people need to do if it's the same for each module, so would be happy with this given your comment and not make any changes, it's lengthy code, but it's transparent code

Reply:
ehm, sorry, :D